contents/security/sqlInjection.md

```diff
@@ -12,37 +12,71 @@
 
-## Overview
+## What is SQL Injection
 
-SQL (Structured Query Language) Injection is a vulnerability found in websites using databases.
-
-SQL is a common language which is used by websites to communicate with databases. Databases can be used to store persistent data, such as usernames and passwords, sensitive account data, or other important information used by the website. Databases are usually made of many "tables" 
-which are organised in rows and columns. Each row is a separate entry in the table, and each column is a specific parameter which can be used by the entry.
-
-As an example, a table in a database may look like this:
+### Background knowledge about SQL
+SQL (Structured Query Language) is a common language which is used by websites to communicate with databases. Databases can be used to store persistent data, such as usernames and passwords, sensitive account data, or other important information used by the website. Databases are usually made of many "tables" which are organised in rows and columns. Each row is a separate entry in the table, and each column is a specific parameter which can be used by the entry. A sample table is shown below:
 
 |   UserId   |    Username   |Password|
 |:----------:|:-------------:|:------:|
 |      1     |     Admin     | 123456 |
 |      2     |     Alice     | pw1234 |
 |      3     |      ...      |   ...  |
 
-
-SQL is used to interact with the database by sending "queries" which the database responds to. Some common SQL commands used include: 
+Then, SQL is used to interact with the database by sending "queries" which the database responds to. Some common SQL commands used include: 
 * `SELECT` - retrieves information from a table
 * `UPDATE` - changes information from a table
 * `INSERT INTO` - adds a new entry into a table
 * `DELETE FROM` - deletes information from a table
+* `DROP` - deletes the whole table
+
+More information about SQL can be found [here](https://www.w3schools.com/sql/).
+
+### SQL Injection
+SQL injection is one of the most common web hacking techniques. It replaces malicious code in SQL statements via web page input. Websites combine SQL syntax with user input to query the underlying database of the website. However, some websites may not check the syntax of user input rigously, and therefore the hacker can inject malicious SQL query via the user input.
+
+## Why is it Important to Prevent SQL Injection
+1. **SQL injection is frequently exhibited.** According to [Open Web Application Security Project(OWASP) report](https://www.owasp.org/images/7/72/OWASP_Top_10-2017_%28en%29.pdf.pdf), the injection attack is always the annual top 1 application security risk from 2013 to 2017. In addition, [Statistics from Akamai](https://www.akamai.com/uk/en/resources/our-thinking/state-of-the-internet-report/web-attack-visualization.jsp) shows that over 80% of attacks are SQL injection in one week.
```

Comment:
put space before "(OWASP)"

Reply:
I think "Statistics from Akamai shows that in one week, over 80% of web attacks are SQL injections." sounds better..